src/snarky.d.ts

```diff
@@ -51,489 +53,767 @@ declare interface ProvablePure<T> extends Provable<T> {
+  assertGt(value: Field | number | string | boolean, message?: string): void;
+  
   /**
    *  @deprecated Deprecated - use {@link assertGreaterThanOrEqual} instead
-   *
-   * Assert that this {@link Field} is greater than or equal to another Field-like value.
-   *
-   * ```ts
-   * Field(1).assertGte(0);
-   * ```
-   *
    */
-  assertGte(y: Field | number | string | boolean, message?: string): void;
+  assertGte(value: Field | number | string | boolean, message?: string): void;
 
   /**
-   * Assert that this {@link Field} equals another Field-like value.
-   * Throws an error if the assertion fails.
-   *
-   * ```ts
-   * Field(1).assertEquals(1);
-   * ```
+   * Assert that this {@link Field} is equal another `field-like` value.
+   * It is equivalent to `Field(...).equals(...).assertEquals(Bool(true))`.
+   * Please see {@link Field.equals} for more details.
+   * 
+   * **Important: If an assertion fails, the code throws an error.
+   * 
+   * @param value - the `field-like` value to compare & assert with this {@link Field}.
+   * @param message? - a string error message to print if the assertion fails, optional.
    */
-  assertEquals(y: Field | number | string | boolean, message?: string): void;
+  assertEquals(value: Field | number | string | boolean, message?: string): void;
 
   /**
-   * Assert that this {@link Field} is either 0 or 1.
-   *
-   * ```ts
-   * Field(0).assertBool();
-   * ```
-   *
+   * Assert that this {@link Field} is equal to 1 or 0 as a `field-like` value.
+   * It is equivalent to `Bool.or(Field(...).equals(1), Field(...).equals(0)).assertEquals(Bool(true))`.
+   * 
+   * **Important: If an assertion fails, the code throws an error.
+   * 
+   * @param value - the `field-like` value to compare & assert with this {@link Field}.
+   * @param message? - a string error message to print if the assertion fails, optional.
    */
   assertBool(message?: string): void;
 
   /**
-   * @deprecated Deprecated - use {@link assertBool} instead
-   *
-   * Assert that this {@link Field} is either 0 or 1.
-   *
-   * ```ts
-   * Field(0).assertBoolean();
-   * ```
-   *
+   * @deprecated Deprecated - use {@link assertBool} instead.
    */
   assertBoolean(message?: string): void;
 
+  /**
+   * Checks if this {@link Field} is 0,
+   * It is equivalent to `Field(...).equals(Field(0))`.
+   * See {@link Field.equals} for more details.
+   * 
+   * @return A {@link Bool} representing if this {@link Field} equals 0.
+   */
   isZero(): Bool;
 
   /**
-   * Little endian binary representation of the field element.
+   * Returns an array of {@link Bool} elements representing [little endian binary representation](https://betterexplained.com/articles/understanding-big-and-little-endian-byte-order/) of this {@link Field} element.
+   * 
+   * **Warning: Binary operations on snarkyJS are currently really costy on chain. Do not use this function if it is not absolutely necessary.
```

Comment:
```suggestion
   * **Warning**: This is a costly operation in a zk proof, because a Field can have 255 bits, each of which has to be constrained to prove that the bits equal the original field element. Prefer to use this method with the optional `length` argument to reduce the maximum amount of bits that the input can have.
```